### PR TITLE
Add New Image button to Work Order Details page

### LINKED ISSUE #26: Add Images (uses built-in camera)
Maybe using something like this: https://github.com/mabelanger/react-html5-camera-photo

Currently uses upload attachment.

screen shot 2018-10-02 at 4 40 11 pm

Most often used for insurance purposes.

### PR DESCRIPTION
**Pull request by mddilley**
Closes #26

### CODE REVIEW

**Comment by mddilley** [comment]
Tested mobile (with vertical camera orientation) on an iPhone. Need to test on Samsung tablet.

File: public/_redirects [hunk 1]
```diff
@@ -0,0 +1 @@
+/* /index.html 200
```
Comment (mateoclarke) [inline public/_redirects]: This seems good? Why did you add it? What does it do?

File: src/components/WorkOrder/AddImage.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    );
+    this.webcam.setup().catch(() => {
+      alert('Error getting access to your camera');
+    });
+  }
+
+  captureImage = async () => {
+    const capturedData = this.webcam.takeBase64Photo({
+      type: 'jpeg',
+      quality: 0.8,
```
Comment (mateoclarke) [inline src/components/WorkOrder/AddImage.js]: Wondering how/if we'd want to tweak this setting...

File: src/queries/api.js [hunk 3]
```diff
@@ -309,7 +336,19 @@ function getHeaders() {
...
     },
-  }
+  };
+}
+
+function getImageHeaders() {
+  return {
+    processData: false,
+    contentType: false,
+    mimeType: `multipart/form-data`,
```
Comment (mateoclarke) [inline src/queries/api.js]: interesting `mimeType`

File: src/styles/AddImage.css.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { css } from 'react-emotion';
```
Comment (mateoclarke) [inline src/styles/AddImage.css.js]: 💅

File: public/_redirects (hunk 1 above)
Comment (mddilley) [reply public/_redirects]: This is a redirect rule that fixed an issue where navigating to the Netlify deploy preview url resulted in a 404 error. This was happening when trying to reach the preview on the Samsung tablet, Sergio's iPad, and my iPhone. My attempt to fix this led me to these resources.

https://github.com/developit/preact-boilerplate/issues/101
https://www.netlify.com/docs/redirects/

File: src/components/WorkOrder/AddImage.js (hunk 2 above)
Comment (mddilley) [reply src/components/WorkOrder/AddImage.js]: Leaving this as is for now. I agree that quality could be lowered to make the app less demanding of bandwidth (high quality jpegs could get large). My initial search about the quality parameter didn't shed much light on exactly how it works.

File: src/styles/AddImage.css.js (hunk 4 above)
Comment (mddilley) [reply src/styles/AddImage.css.js]: This is a cool library. Very similar to StyledComponents, and I like it. 👩‍🎤

**Comment by mddilley** [comment]
Thanks for the props on this PR! It was super fun to dive into the unknown on this one. There is still much to learn about the `MediaDevices.getUserMedia()` HTML 5 method, but I did find a way to turn off the webcam stream. I added a new method to the Camera class and called it in `componentWillUnmount()` so no more endless 👀.
